src/npc_class.cpp

```diff
@@ -398,7 +338,7 @@ const npc_class_id &npc_class::random_common()
     }
 
     if( common_classes.empty() || one_in( common_classes.size() ) ) {
-        return NC_NONE;
+        return npc_class_id::NULL_ID();
```

Comment:
Does npc_class_id::NULL_ID() actually resolve to NC_NONE? NC_NONE is bizarrely a completely valid class which *sometimes shows up on purpose(?!)*

If it doesn't then randomized NPCs who spawn and receive this null_id class will throw an error. Maybe we should just remove the `one_in( common_classes.size() )` bit.

Reply:
The null id is NC_NONE so this should behave identically
https://github.com/CleverRaven/Cataclysm-DDA/blob/daea5c2177089e7f49b48e0fbd1d229c7488aba2/src/string_id_null_ids.cpp#L35